File: mycollab-config/src/main/java/com/esofthead/mycollab/configuration/EmailConfiguration.java
/**
 * This file is part of mycollab-config.
 *
 * mycollab-config is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * mycollab-config is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with mycollab-config.  If not, see <http://www.gnu.org/licenses/>.
 */
package com.esofthead.mycollab.configuration;

/**
 * Email configuration of MyCollab
 * 
 * @author MyCollab Ltd.
 * @since 1.0
 * 
 */
public class EmailConfiguration {
	private String host;
	private String user;
	private String password;
	private int port;
	private boolean isTls;

	EmailConfiguration(String host, String username, String password, int port,
			boolean isTLS) {
		this.host = host;
		this.user = username;
		this.password = password;
		this.port = port;
		this.isTls = isTLS;
	}

	public String getHost() {
		return host;
	}

	public String getUser() {
		return user;
	}

	public String getPassword() {
		return password;
	}

	public int getPort() {
		return port;
	}

	public boolean isTls() {
		return isTls;
	}

}

File: mycollab-config/src/main/java/com/mycollab/configuration/EmailConfiguration.kt
/**
 * Copyright © MyCollab
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http:></http:>//www.gnu.org/licenses/>.
 */
package com.mycollab.configuration

import com.mycollab.core.arguments.ValuedBean
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.Profile
import org.springframework.stereotype.Component

/**
 * Email configuration of MyCollab
 *
 * @author MyCollab Ltd.
 * @since 1.0
 */
@Component
@Profile("program")
@ConfigurationProperties(prefix = "mail")
open class EmailConfiguration(var smtphost: String?, var username: String?,
                              var password: String?, var port: Int,
                              var startTls: Boolean = false, var ssl: Boolean = false,
                              var notifyEmail: String) : ValuedBean() {
    constructor(): this("", "", "", -1, false, false, "")
}


